### PR TITLE
fix(imagebrush): downscaling images was very blurry

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs [hunk 1]
```diff
@@ -79,13 +79,33 @@ internal override void UpdatePaint(SKPaint fillPaint, SKRect bounds)
...
+
+				// Adding image downscaling in the shader matrix directly is very blurry
+				// so we use this workaround instead.
+				// https://github.com/mono/SkiaSharp/issues/520#issuecomment-444973518
+				var kernel = new[]
+				{
+					0, -.1f, 0,
+					-.1f, 1.4f, -.1f,
+					0, -.1f, 0,
+				};
```
Comment (Youssef1313) [inline src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs:92]: This should be a private field like:

```csharp
private ReadOnlySpan<float> Kernel => [0, -0.1f, 0, -0.1f, 1.4f, -0.1f, 0, -0.1f, 0];
```

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI.Composition/Composition/CompositionSpriteShape.skia.cs [hunk 2]
```diff
@@ -126,6 +126,12 @@ private static SKPaint TryCreateAndClearPaint(in Visual.PaintingSession session,
 					paint.PathEffect.Dispose();
 					paint.PathEffect = null;
 				}
+
+				if (paint.ImageFilter != null)
```
Comment (jeromelaban) [inline src/Uno.UI.Composition/Composition/CompositionSpriteShape.skia.cs:130]: Is this something that can be tested? A simple screenshot validation can do the trick.

**Comment by ramezgerges** [comment]
The initial fix broke brushes that use ImageFilters (e.g. Acrylic brushes) and is overall not very good-looking. I simplified the fix to just force Lanczos.

Before
After
image

**Comment by jeromelaban** [comment]
@ramezgerges the change is causing [other tests to fail](https://dev.azure.com/uno-platform/Uno%20Platform/_TestManagement/Runs?runId=2886460&_a=resultSummary&resultId=100752), could you take a look?

**Pull request merged**